leshan-lwm2m-core/src/main/java/org/eclipse/leshan/core/node/codec/json/LwM2mNodeJsonEncoder.java

```diff
@@ -263,61 +263,68 @@ private ArrayList<JsonArrayEntry> lwM2mResourceToJsonArrayEntry(String resourceP
         private JsonArrayEntry createJsonArrayEntry(String name, BigDecimal timestampInSeconds, Type type,
                 Type expectedType, Object value) {
             // Create resource element
-            JsonArrayEntry jsonResourceElt = new JsonArrayEntry();
-            jsonResourceElt.setName(name);
-            jsonResourceElt.setTime(timestampInSeconds);
+            JsonArrayEntry jsonResourceElt = new JsonArrayEntry(name, null, null, null, null, timestampInSeconds);
 
             // Convert value using expected type
             LwM2mPath lwM2mResourcePath = name != null ? new LwM2mPath(name) : null;
             Object convertedValue = converter.convertValue(value, type, expectedType, lwM2mResourcePath);
-            this.setResourceValue(convertedValue, expectedType, jsonResourceElt, lwM2mResourcePath);
+            jsonResourceElt = this.setResourceValue(convertedValue, expectedType, jsonResourceElt, lwM2mResourcePath);
 
             return jsonResourceElt;
         }
 
-        private void setResourceValue(Object value, Type type, JsonArrayEntry jsonResource, LwM2mPath resourcePath) {
+        private JsonArrayEntry setResourceValue(Object value, Type type, JsonArrayEntry jsonResource,
+                LwM2mPath resourcePath) {
             LOG.trace("Encoding value {} in JSON", value);
 
             if (type == null || type == Type.NONE) {
                 throw new CodecException(
                         "Unable to encode value for resource {} without type(probably a executable one)", resourcePath);
             }
 
+            String name = jsonResource.getName();
+            BigDecimal time = jsonResource.getTime();
+            Number floatValue = jsonResource.getFloatValue();
+            Boolean booleanValue = jsonResource.getBooleanValue();
+            String stringValue = jsonResource.getStringValue();
+            String objectLinkValue = jsonResource.getObjectLinkValue();
+
             // Following table 20 in the Specs
             switch (type) {
             case STRING:
-                jsonResource.setStringValue((String) value);
+                stringValue = (String) value;
                 break;
             case INTEGER:
             case UNSIGNED_INTEGER:
             case FLOAT:
-                jsonResource.setFloatValue((Number) value);
+                floatValue = (Number) value;
                 break;
             case BOOLEAN:
-                jsonResource.setBooleanValue((Boolean) value);
+                booleanValue = (Boolean) value;
                 break;
             case TIME:
                 // Specs device object example page 44, rec 13 is Time
                 // represented as float?
-                jsonResource.setFloatValue((((Date) value).getTime() / 1000L));
+                floatValue = BigDecimal.valueOf((((Date) value).getTime() / 1000L));
```

Comment:
Here, there is a behavior change.

I mean before we created a Long and now a BigDecimal.

Is it intended ? 
If yes, what is the purpose of that ?